src/components/app/Watchlist.svelte

```diff
@@ -0,0 +1,320 @@
+<script lang="typescript">
+  import { watchlist } from "../../stores/watchlistStore.ts";
+  import editIcon from "../../assets/edit.svg";
+  import closeIcon from "../../assets/close.svg";
+  import addIcon from "../../assets/add.svg";
+  import Line from "svelte-chartjs/src/Line.svelte";
+  import { fade, scale } from "svelte/transition";
+  import type { IWatchlistElement } from "../../utils/types.ts";
+  import Modal from "../Modal.svelte";
+  import { notification } from "../../stores/notificationStore.ts";
+  import { NotificationType } from "../../utils/types.ts";
+
+  import { onMount } from "svelte";
+
+  import Verto from "@verto/lib";
+  const client = new Verto();
+
+  let tokens;
+  onMount(async () => {
+    tokens = await client.getTokens();
+    for (const element of $watchlist) {
+      const index = tokens.indexOf(
+        tokens.find((token) => token.ticker === element.ticker)
+      );
+      if (index > -1) {
+        tokens.splice(index, 1);
+      }
+    }
+  });
+
+  let editMode = false;
+  let addModalOpened = false;
+  let addPst;
+  let addPeriod: number = 7;
+
+  function add() {
+    const element = tokens.find((token) => token.ticker === addPst);
+    const index = tokens.indexOf(element);
+    tokens.splice(index, 1);
+
+    watchlist.addPst({
+      id: element.id,
+      name: element.name,
+      ticker: element.ticker,
+      period: addPeriod,
+    });
+
+    addPeriod = 24;
+    notification.notify(
+      "Added",
+      `Added ${element.ticker} to watchlist.`,
+      NotificationType.success,
+      2000
+    );
+
+    addPst = tokens[0] && tokens[0].ticker;
+  }
+
+  function remove(pst: IWatchlistElement) {
+    tokens.push({
+      id: pst.id,
+      name: pst.name,
+      ticker: pst.ticker,
+    });
+    watchlist.removePst(pst.ticker);
+    notification.notify(
+      "Removed",
+      `Removed ${pst.ticker} from watchlist.`,
+      NotificationType.error,
+      2000
+    );
+  }
+
+  function openAddModal() {
+    if (tokens.length === 0) {
+      notification.notify(
+        "Error",
+        `No more PSTs available to add.`,
+        NotificationType.error,
+        1000
+      );
+      return;
+    }
+    addModalOpened = true;
+  }
+
+  async function load(token: string, period: number) {
+    const returnedPrices = await client.price(token);
+    const length = returnedPrices.prices.length - 1;
+
+    const dates = returnedPrices.dates.slice(length - period, length);
+    const prices = returnedPrices.prices.slice(length - period, length);
+
+    const start = prices[0];
+    const end = prices[prices.length - 1];
+    let color;
+    let percentage = (end - start) / start;
```

Comment:
For percent change, I think you'll need to multiply this by 100 to get the actual percentage. I did the math to make sure this works, and it checks out.

Reply:
@t8 Yup 😅 That's absolutely correct. Fixed in https://github.com/useverto/verto/pull/91/commits/b34b5f2d6a60f4c576551995a7d1c8be91e3f61d.